### ISSUE TITLE
Implement ability to monitor the value of feature x to show / hide dynamically

### ISSUE THREAD

**Issue by ScottSpittle** [willmendesneto/ngx-feature-toggle#81]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
```

**Current behavior**
n/a

**Expected behavior**
n/a

**Reproduction of the problem**
n/a

**What is the motivation / use case for changing the behavior?**
It would be nice if the toggle states were monitored and the element could show/hide depending on that state.

i.e element x is visible, API call goes off and returns a 503, then element x is not visible

**Please tell us about your environment:**
n/a

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
n/a

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (if applicable):** `node --version` =

**Comment by willmendesneto** [comment]
Hi, @ScottSpittle .

Thanks for your request. I implemented the feature to listener the changes [in this commit](https://github.com/willmendesneto/ngx-feature-toggle/commit/7669faaaa72b334db2e67e9e8ffc86931fad353f). I added `ngDoCheck` + `ngZone` for this, after some check some performance benchmark between approaches:

- `ngDoCheck` only:

<img width="1024" alt="screen shot 2017-11-05 at 12 34 30 pm" src="https://user-images.githubusercontent.com/1252570/32414882-c8d86966-c283-11e7-8720-d1f30184bf7f.png">


- `ngDoCheck` + `ngZone`:
<img width="1014" alt="screen shot 2017-11-05 at 12 35 15 pm" src="https://user-images.githubusercontent.com/1252570/32414883-c906b1ae-c283-11e7-8818-4bb49855e7b8.png">

As you can see, there's a huge difference between approaches when you check the frames and functions on Main thread.

I'm using this approach, so that the consumers who are using this component can update the data using different approaches.

In you example, you can change the values of `<feature-toggle-provider>` component when you have a HTTP call with a non-expected response, as example.

Let me know if this issue should be closed or not.

**Comment by willmendesneto** [comment]
UPDATED:

This feature was released on v4.1.0 🎉  🎉  🎉

**Comment by willmendesneto** [comment]
Closing this issue by lack on the response. Thanks @ScottSpittle
